sycl/doc/Array_Kernel_Parameters.md

```diff
@@ -0,0 +1,359 @@
+<h2>Array Parameters of SYCL Kernels</h2>
+
+<h3>Introduction</h3>
+
+This document describes the changes to support passing arrays to SYCL kernels and special treatment of Accessor arrays. The following cases are handled:
+
+1. arrays of standard-layout type as top-level arguments
+2. arrays of Accessors as top-level arguments
+3. arrays of accessors within structs that are top-level arguments
+
+The motivation for this correction to kernel parameters processing is to bring consistency to the treatment of arrays. On the CPU, a lambda function is allowed to access an element of an array defined outside the lambda. The implementation captures the entire array by value. A user would naturally expect this to work in SYCL as well. However, the current implementation flags references to arrays defined outside a SYCL kernel as errors.
+
+The first few sections describe the current design. The last three sections describe the design to support 1. to 3. above. The implementation of this design is confined to three functions in the file SemaSYCL.cpp.

```

Comment:
```suggestion
The first few sections describe the current design. The last three sections describe the design to support 1. to 3. above. The implementation of this design is confined to three functions in the file `SemaSYCL.cpp`.
```
More generally there are a lot of lacking backticks in the following paragraphs.

Reply:
Done.